packages/components/src/Card/Card.tsx

```diff
@@ -4,23 +4,49 @@ import { XOR } from "ts-xor";
 import styles from "./Card.css";
 import colors from "./colors.css";
 import { CardClickable } from "./CardClickable";
+import { Button, ButtonProps } from "../Button";
 import { Heading } from "../Heading";
 
-interface CardProps {
+export type HeaderProps = XOR<HeaderActionProps, HeaderCustomProps>;
+interface CardCommonProps {
   /**
    * The `accent`, if provided, will effect the color accent at the top of
    * the card.
    */
   readonly accent?: keyof typeof colors;
   readonly children: ReactNode | ReactNode[];
+}
+
+interface CardHeaderProps {
+  readonly title?: never;
+
+  /**
+   * The header props of the card.
+   */
+  readonly header?: HeaderProps;
+}
 
+interface CardTitleProps {
   /**
-   * The title of the card.
+   * @deprecated Use header instead
    */
   readonly title?: string;
+
+  readonly header?: never;
+}
+
+type CardProps = CardCommonProps & XOR<CardHeaderProps, CardTitleProps>;
```

Comment:
2 things are confusing me while reading the types. I think they are pre-existing issues that extra types have made more confusing.

1. The order of the types. Normally I like defining lower level interfaces first and use them below. In here we seem to have a mix of defining an interface using other interfaces that are defined both above and below. It would be nice if we could do one or the other. 
2. The names are a little tricky to follow, though I am not sure how much can be done about that.

Reply:
Actually, on the names it might help to reduce the number of named types. If we don't need to export the `HeaderProps` which I don't think we do, you could not create it and instead assign header directly.

```
interface CardHeaderProps {
  readonly title?: never;

  /**
   * The header props of the card.
   */
  readonly header?: XOR<HeaderActionProps, HeaderCustomProps>;
}
```

Reply:
the `HeaderProps` type is used in `checkHeaderVisibility(...)`, but the `export` is not required